app/server/endpoint/src/main/java/io/syndesis/server/endpoint/v1/handler/connection/ConnectionHandler.java

```diff
@@ -133,15 +136,43 @@ public ConnectionOverview get(final String id) {
         final ConnectionOverview.Builder builder = new ConnectionOverview.Builder().createFrom(connection);
 
         // set the connector
-        DataManagerSupport.fetch(dataManager, Connector.class, connection.getConnectorId())
-            .ifPresent(builder::connector);
+        Optional<Connector> c = DataManagerSupport.fetch(dataManager, Connector.class, connection.getConnectorId());
+        enrichWithDynamicProperties(c).ifPresent(builder::connector);
 
         // set the board
         DataManagerSupport.fetchBoard(dataManager, ConnectionBulletinBoard.class, id).ifPresent(builder::board);
 
         return builder.build();
     }
 
+    private static Optional<Connector> enrichWithDynamicProperties(Optional<Connector> conn) {
+        if (conn != null && conn.isPresent()) {
```

Comment:
Prefer early exit:
```suggestion
        if (conn == null || !conn.isPresent()) {
            return Optional.empty();
        }
```

Reply:
Make sense. Waiting for @gashcrumb to take a look at the UI side and check if this PR is really needed. If it is, I'm moving that return upwards.

Reply:
I think this PR is definitely needed, on staging the call to fetch the actions doesn't return the `defaultValue` property filled in, just verified that.  Still getting up to speed as to why it may not show up in the UI.

Reply:
In fact, I would propose if this gets the `defaultValue` set properly in the response, let's get it in and we can follow up with an additional PR with a fix to the UI if needed.

Reply:
After looking at this further I'm a bit concerned that your fix is being masked by another issue that's crept into the codebase.  I've raised https://github.com/syndesisio/syndesis/issues/8498 to split it out so I stop commenting in your PR :-)

Reply:
nope, the problem is right here in front of my face.  [This bit](https://github.com/syndesisio/syndesis/blob/master/app/ui-react/syndesis/src/modules/integrations/components/editor/endpoint/WithConfigurationForm.tsx#L142) is ensuring that for webhook specifically we don't query the BE, because we hit that case, override the definition and then return the cached descriptor vs fetch it from the BE.  @kahboom @KurtStam hey, aren't you guys looking at an issue in this area also?

Reply:
But anyway, we should push this, right?

Reply:
One more thing, check for `x != null` should not be done on a method that receives `Optional`, that's the whole point of `Optional`...

Reply:
I also wonder that, I just don't trust the optional can't be null :D

Removing the null check then.

Reply:
Hey, just to follow up, that line I pointed to is also the cause of ENTESB-13527, because normally when that form is shown the UI *always* fetches the descriptor for the action as part of showing the form.  Adding the extra conditional for webhook skipped that.  So yeah, this change isn't needed as the UI currently only uses the API response in this case to list the available actions for the user to pick one.

Reply:
Then closing and not merging :)

Let me know if something else is needed!